### PR TITLE
fix: synced WinState was causing issues on a replay attempt (MTT-4490)

### PR DESCRIPTION
**Pull request by pdeschain**
### Description

Was:
 - BossRoom scene pickled a networkObject with WinState netvar with DontDestroyOnLoad
 - PostGame scene was supposed to clean up that networkObject
 - however when BossRoom scene is exited to MainMenu - that cleanup didn't happen, causing issues on an attempt to enter the BossRoom scene again

Now:
 - BossRoom state passes the WinState via a new server-only PersistentGameState (plain C# class, passed in via DI)
 -  PostGame state is loaded, and on the server it passes the data from persistentGameState to a new NetworkBehaviour called PostGameStateData (yes, it's not a very pretty name) and it then in turn synchronizes the WinState information across all the peers. 
 - This way we do not need to pickle networkObjects with DontDestroyOnLoad
- Late joiners are automatically taken care of by a netvar in PostGameStateData, which in turn doesn't survive beyond the PostGame scene

### Issue Number(s)

MTT-4490

### Contribution checklist
 - [ ] Tests have been added for boss room and/or utilities pack
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file and/or [package changelog](../Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ Additional documentation and release notes are available at [Multiplayer Documen
 ### Fixed
 * Subscribing to a message channel while unsubscribing is pending (#675)
 * Using ```Visible``` instead of ```Enabled``` to make sure RNSM continues updating when off (#702)
+* Fixed a bug caused by DontDestroyOnLoad of a networkbehaviour that carried a WinState netvar - now this state is passed by server to the PostGame scene and it then stores that state in the netvar, eliminating the need to preserve a NetworkBehaviour-bearing gameObject across scenes. (#724)
```
Comment (SamuelBellomo) [inline CHANGELOG.md:36]: Was this bug a thing in 1.3.0?

**Review by SamuelBellomo — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]
Looks like this is fixing MTT-4487 as well

**Review by fernando-cortez — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pdeschain) [reply CHANGELOG.md:38]: not sure if it was released actually

Comment (fernando-cortez) [reply CHANGELOG.md:38]: This wasn't an issue in 1.3.0, so this blurb can be removed since we've introduced it since, right?

Comment (pdeschain) [reply CHANGELOG.md]: done! rephrased and moved to Changed

**Review by SamuelBellomo — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**